src/overlays/actors/ovl_Obj_Hsblock/z_obj_hsblock.c

```diff
@@ -32,36 +35,120 @@ const ActorInit Obj_Hsblock_InitVars = {
+    this->actionFunc = actionFunc;
+}
+
+void func_8093DEAC(ObjHsblock* this, PlayState* play) {
+    if (OBJHSBLOCK_GET_5(&this->dyna.actor) != 0) {
+        Actor_SpawnAsChild(&play->actorCtx, &this->dyna.actor, play, ACTOR_OBJ_ICE_POLY, this->dyna.actor.world.pos.x,
+                           this->dyna.actor.world.pos.y, this->dyna.actor.world.pos.z, this->dyna.actor.world.rot.x,
+                           this->dyna.actor.world.rot.y, this->dyna.actor.world.rot.z, 0xFF64);
+    }
+}
+
+void ObjHsblock_Init(Actor* thisx, PlayState* play) {
+    ObjHsblock* this = THIS;
+
+    DynaPolyActor_Init(&this->dyna, 0);
+    DynaPolyActor_LoadMesh(play, &this->dyna, sColHeaders[OBJHSBLOCK_GET_3(thisx)]);
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    func_8093DEAC(this, play);
+
+    switch (OBJHSBLOCK_GET_3(&this->dyna.actor)) {
+        case 0:
+        case 2:
+            func_8093E03C(this);
+            break;
+        case 1:
+            if (Flags_GetSwitch(play, OBJHSBLOCK_GET_SWITCH(thisx))) {
+                func_8093E03C(this);
+            } else {
+                func_8093E05C(this);
+            }
+            break;
+        default:
+            break;
+    }
+}
+
+void ObjHsblock_Destroy(Actor* thisx, PlayState* play) {
+    ObjHsblock* this = THIS;
+
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+}
+
+void func_8093E03C(ObjHsblock* this) {
+    ObjHsblock_SetupAction(this, NULL);
+}
+
+void func_8093E05C(ObjHsblock* this) {
+    this->dyna.actor.flags |= ACTOR_FLAG_10;
+    this->dyna.actor.world.pos.y = this->dyna.actor.home.pos.y - 105.0f;
+    ObjHsblock_SetupAction(this, func_8093E0A0);
+}
+
+void func_8093E0A0(ObjHsblock* this, PlayState* play) {
+    if (Flags_GetSwitch(play, OBJHSBLOCK_GET_SWITCH(&this->dyna.actor))) {
+        func_8093E0E8(this);
+    }
+}
+
+void func_8093E0E8(ObjHsblock* this) {
+    ObjHsblock_SetupAction(this, func_8093E10C);
+}
+
+void func_8093E10C(ObjHsblock* this, PlayState* play) {
+    Math_SmoothStepToF(&this->dyna.actor.velocity.y, 16.0f, 0.1f, 0.8f, 0.0f);
+    if (fabsf(Math_SmoothStepToF(&this->dyna.actor.world.pos.y, this->dyna.actor.home.pos.y, 0.3f,
+                                 this->dyna.actor.velocity.y, 0.3f)) < 0.001f) {
+        this->dyna.actor.world.pos.y = this->dyna.actor.home.pos.y;
+        func_8093E03C(this);
+        this->dyna.actor.flags &= ~ACTOR_FLAG_10;
+    }
+}
+
+void ObjHsblock_Update(Actor* thisx, PlayState* play) {
+    ObjHsblock* this = THIS;
+
+    if (this->actionFunc != NULL) {
+        this->actionFunc(this, play);
+    }
+    Actor_SetFocus(&this->dyna.actor, sFocusPoss[OBJHSBLOCK_GET_3(thisx)]);
+}
+
+void ObjHsblock_Draw(Actor* thisx, PlayState* play) {
+    static Color_RGB8 sEnvColors[] = {
+        { 60, 60, 120 },
+        { 120, 100, 70 },
+        { 100, 150, 120 },
+        { 255, 255, 255 },
+    };
+    Color_RGB8* envColors = &sEnvColors[OBJHSBLOCK_GET_6(thisx)];
```

Comment:
```suggestion
    Color_RGB8* envColor = &sEnvColors[OBJHSBLOCK_GET_6(thisx)];
```
Only grabbing a single color from the array.